alf/trainers/on_policy_trainer.py

```diff
@@ -12,259 +12,41 @@
-        checkpoint_name (str): name of the checkpoint (e.g. 'ckpt-12800`).
-            If None, the latest checkpoint unber train_dir will be used.
-        greedy_predict (bool): use greedy action for evaluation.
-        random_seed (int): random seed
-        num_episodes (int): number of episodes to play
-        sleep_time_per_step (float): sleep so many seconds for each step
-        record_file (str): if provided, video will be recorded to a file
-            instead of shown on the screen.
-        use_tf_functions (bool): whether to use tf.function
-    """
-    root_dir = os.path.expanduser(root_dir)
-    train_dir = os.path.join(root_dir, 'train')
-
-    tf.random.set_seed(random_seed)
-    global_step = get_global_counter()
-
-    driver = OnPolicyDriver(
-        env=env,
-        algorithm=algorithm,
-        training=False,
-        greedy_predict=greedy_predict)
-
-    ckpt_dir = os.path.join(train_dir, 'algorithm')
-    checkpoint = tf.train.Checkpoint(
-        algorithm=algorithm,
-        metrics=metric_utils.MetricsGroup(driver.get_metrics(), 'metrics'),
-        global_step=global_step)
-    if checkpoint_name is not None:
-        ckpt_path = os.path.join(ckpt_dir, checkpoint_name)
-    else:
-        ckpt_path = tf.train.latest_checkpoint(ckpt_dir)
-    if ckpt_path is not None:
-        logging.info("Restore from checkpoint %s" % ckpt_path)
-        checkpoint.restore(ckpt_path)
-    else:
-        logging.info("Checkpoint is not found at %s" % ckpt_dir)
-
-    if not use_tf_functions:
-        tf.config.experimental_run_functions_eagerly(True)
-
-    recorder = None
-    if record_file is not None:
-        recorder = VideoRecorder(env.pyenv.envs[0], path=record_file)
-    else:
-        # pybullet_envs need to render() before reset() to enable mode='human'
-        env.pyenv.envs[0].render(mode='human')
-    env.reset()
-    if recorder:
-        recorder.capture_frame()
-    time_step = driver.get_initial_time_step()
-    policy_state = driver.get_initial_policy_state()
-    episode_reward = 0.
-    episode_length = 0
-    episodes = 0
-    while episodes < num_episodes:
-        time_step, policy_state = driver.run(
-            max_num_steps=1, time_step=time_step, policy_state=policy_state)
-        if recorder:
-            recorder.capture_frame()
-        else:
-            env.pyenv.envs[0].render(mode='human')
-            time.sleep(sleep_time_per_step)
-        if time_step.is_last():
-            logging.info("episode_length=%s episode_reward=%s" %
-                         (episode_length, episode_reward))
-            episode_reward = 0.
-            episode_length = 0.
-            episodes += 1
-        else:
-            episode_reward += float(time_step.reward)
-            episode_length += 1
-    if recorder:
-        recorder.close()
-    env.reset()
+class OnPolicyTrainer(Trainer):
+    def __init__(self,
+                 root_dir,
+                 train_interval=20,
```

Comment:
we can change to 'unroll_length' to be consistent with off_policy_trainer.

Reply:
done